### DISCUSSION TITLE
Session Issue With Firefox

### DISCUSSION THREAD

**Discussion by pschiffe** [ngoduykhanh/PowerDNS-Admin#1749] [Q&A] [answered]
Hello,

I've tried both recent versions - `v0.4.1` and `v0.4.2` with clean database and `mariadb:11` as sql server. In the current Firefox version (`v122.0`), the web ui won't work and throws an 500 error. In the logs I see the following:
```
sqlalchemy.exc.DataError: (MySQLdb._exceptions.DataError) (1406, "Data too long for column 'session_id' at row 1")
[SQL: INSERT INTO sessions (session_id, data, expiry) VALUES (%s, %s, %s)]
[parameters: ('session:.eJxtkEtvGzEMhP9KoFxTQA9SD9-cOkB6Tns2KIm0F9nsFmstWsDwf482RdAeeiJIzDcYzlUd-XdbSO2UCRoQU3IIxqgHdZSFL2e1a8vKfRvqpkkVpEZEqTVHF6uVgLYAVEjW1kJAjnUy ... (196 characters truncated) ... 3TpSLosc29wD9XsKUGxKMWfdCzNZGwPRFebKYnL2AhoRBLrTSNNppRN3irdPx_k0TP983uj0UcVV3a2bMwg60mwMRqDiYyWXxEJAJCQtPcw7MeyS9Q.ZZmLgQ.NXT0u-D9HM5obNEwHEPvDq-BAtc', b'\x80\x04\x95y\x00\x00\x00\x00\x00\x00\x00}\x94(\x8c\n_permanent\x94\x88\x8c\x06_fresh\x94\x89\x8c\x04next\x94\x8c\x01/\x94\x8c\x0b_csrf_token\x94\x8c@57d0f03da9fca83cda928c9dfd6c1a2c33aa1949f061c9ac7ea857d7e40f8108\x94u.', datetime.datetime(2024, 2, 1, 16, 49, 21, 406800))]
```

`session_id` column definition is [here](https://github.com/PowerDNS-Admin/PowerDNS-Admin/blob/master/powerdnsadmin/models/sessions.py#L8) and should be 255 characters long, which I can confirm in phpmyadmin. I don't know why it seems that Firefox has longer session ID. In the previous Firefox versions this worked and the pdns admin `v0.4.1` works in Chrome as well.

In the pdns admin configuration I have `SESSION_TYPE = 'sqlalchemy'`.

**Comment by pschiffe** [comment]
Increasing `session_id` length to `varchar(511)` seems to fix the problem.
```
MariaDB [powerdnsadmin]> SELECT `session_id` FROM `sessions` WHERE `id` = 398;
session:.eJxtkEtvGzEMhP9KoFxTQA9SD9-cOkB6Tns2KIm0F9nsFmstWsDwf482RdAeeiJIzDcYzlUd-XdbSO2UCRoQU3IIxqgHdZSFL2e1a8vKfRvqpkkVpEZEqTVHF6uVgLYAVEjW1kJAjnUyJLG4nGwMDg3HiKZEnU2MGa2HEIoln312Ur2tbLXz1hJkH7UYhIhetDfSbYMEwayRsZaEssVaL7z8SUP1bZj6idZ25qkNhRrXz8CZyitPm-xx__Lt627_4_tz15Z5nJeL2l1VXpefn_NLGYfusK0fgk7d68MB_aEjrTfUD3nsjur2oMa50PgfwGp4jE9_gV_nobG63TpSLosc29wD9XsKUGxKMWfdCzNZGwPRFebKYnL2AhoRBLrTSNNppRN3irdPx_k0TP983uj0UcVV3a2bMwg60mwMRqDiYyWXxEJAJCQtPcw7MeyS9Q.ZZmLgQ.NXT0u-D9HM5obNEwHEPvDq-BAtc
```

**Comment by AzorianMatt** [comment]
@pschiffe aren't you building your own images though versus using the project's images? Only raising this concern because some of the reports we have previously received, have come back to dated releases on your end at one time in the past.

No idea if it's relevant, but I know there is an update out to the relevant dependency that manages this (flask-session). It has not been rolled into the code base. I know this will be an important update because it appears to address to previously known bugs that affected this project.

**Comment by pschiffe** [comment, marked as answer]
Hi @AzorianMatt, thank you for your reply. Today I wanted  to take another look at this issue, but I cannot reproduce it anymore :shrug: 
I did update and restart my computer (Fedora Linux) in the meantime, which may or may not be related, but knowing Linux, it probably was.

Anyway, today Firefox and Chrome seem to work the same way, and even though I still have some session issues similar to some other thread in the discussions. I hope the `flask-session` update will fix it and I'm looking forward to the next `v0.4.3` version :crossed_fingers: 

And yeah - I'm sorry about the outdated image, it should be better now.

**Reply by AzorianMatt** [reply]
@pschiffe you're welcome and thank you for reaching out as well. Sounds like a typical day in the life of!

On that update, feel free to try it out if you want, it's in the `dev` branch now. No worries on the images, I see you are much more active on them now.
